connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaConfigBackingStore.java

```diff
@@ -547,6 +663,27 @@ KafkaBasedLog<String, byte[]> setupAndCreateKafkaBasedLog(String topic, final Wo
         return createKafkaBasedLog(topic, producerProps, consumerProps, new ConsumeCallback(), topicDescription, adminSupplier);
     }
 
+    private void maybeSendFencably(String key, byte[] value) {
+        if (!usesFencableWriter) {
+            configLog.send(key, value);
+            return;
+        }
+
+        if (fencableProducer == null) {
+            throw new IllegalStateException("Cannot produce to config topic without claiming write privileges first");
+        }
+
+        try {
+            fencableProducer.beginTransaction();
+            fencableProducer.send(new ProducerRecord<>(topic, key, value));
+            fencableProducer.commitTransaction();
+        } catch (ProducerFencedException e) {
+            log.warn("Failed to write to config topic as producer was fenced out", e);
+            fencableProducer = null;
```

Comment:
Should we close the `fencableProducer` when `ProducerFencedException` thrown? Looks like we only recreate one afterwards, right? And what about other exceptions? Should we also close `fencableProducer` in those cases?

Reply:
Yes, we should close it in this case. I originally wanted to be more conservative about other cases, but after thinking it over, I'd rather not risk getting into a bad state with a transactional producer that can't be fixed and requires a new one. Given the expected low frequency of writes to the config topic, it's probably fine to also close the producer and then construct a new one in `claimWritePrivileges` in this case. Good catch!